src/notation/internal/notationactioncontroller.cpp

```diff
@@ -1907,6 +1920,110 @@ void NotationActionController::toggleStrike()
     currentNotationInteraction()->toggleStrike();
 }
 
+bool NotationActionController::isMasterNotation(const INotationPtr notation) const
+{
+    return globalContext()->currentMasterNotation()->notation() == notation;
+}
+
+QList<INotationPtr> NotationActionController::getOpenNotations()
+{
+    QList<INotationPtr> notations;
+    auto excerpts = globalContext()->currentMasterNotation()->excerpts().val;
+    for (IExcerptNotationPtr excerpt: excerpts) {
+        INotationPtr notation = excerpt->notation();
+        if (notation->opened().val) {
+            notations.append(notation);
+        }
+    }
+    return notations;
+}
+
+void NotationActionController::closeCurrentNotation()
+{
+    INotationPtr nextNotation = nullptr;
+    IMasterNotationPtr masterNotation = globalContext()->currentMasterNotation();
+    QList<INotationPtr> notations = getOpenNotations();
+    for (int i = 0; i < notations.size(); i++) {
+        INotationPtr notation = notations[i];
```

Comment:
If you do not want to modify the value in the list, then it is better to use the `at` method